Connectors/Autocad/Speckle.Connectors.AutocadShared/Operations/Receive/AutocadHostObjectBaseBuilder.cs

```diff
@@ -0,0 +1,288 @@
+  private readonly RootObjectUnpacker _rootObjectUnpacker;
+  private readonly IReceiveConversionHandler _conversionHandler;
+
+  protected AutocadHostObjectBaseBuilder(
+    IRootToHostConverter converter,
+    AutocadLayerBaker layerBaker,
+    AutocadGroupBaker groupBaker,
+    AutocadInstanceBaker instanceBaker,
+    IAutocadMaterialBaker materialBaker,
+    IAutocadColorBaker colorBaker,
+    AutocadContext autocadContext,
+    RootObjectUnpacker rootObjectUnpacker,
+    IReceiveConversionHandler conversionHandler
+  )
+  {
+    _converter = converter;
+    _layerBaker = layerBaker;
+    _groupBaker = groupBaker;
+    _instanceBaker = instanceBaker;
+    _materialBaker = materialBaker;
+    _colorBaker = colorBaker;
+    _autocadContext = autocadContext;
+    _rootObjectUnpacker = rootObjectUnpacker;
+    _conversionHandler = conversionHandler;
+  }
+
+  public Task<HostObjectBuilderResult> Build(
+    Base rootObject,
+    string projectName,
+    string modelName,
+    IProgress<CardProgress> onOperationProgressed,
+    CancellationToken cancellationToken
+  )
+  {
+    // Prompt the UI conversion started. Progress bar will swoosh.
+    onOperationProgressed.Report(new("Converting", null));
+
+    // Layer filter for received commit with project and model name
+    _layerBaker.CreateLayerFilter(projectName, modelName);
+
+    // 0 - Clean then Rock n Roll!
+    string baseLayerPrefix = _autocadContext.RemoveInvalidChars($"SPK-{projectName}-{modelName}-");
+    PreReceiveDeepClean(baseLayerPrefix);
+
+    // 1 - Unpack objects and proxies from root commit object
+    var unpackedRoot = _rootObjectUnpacker.Unpack(rootObject);
+
+    // 2 - Split atomic objects and instance components with their path
+    var (atomicObjects, instanceComponents) = _rootObjectUnpacker.SplitAtomicObjectsAndInstances(
+      unpackedRoot.ObjectsToConvert
+    );
+    var atomicObjectsWithPath = _layerBaker.GetAtomicObjectsWithPath(atomicObjects);
+    var instanceComponentsWithPath = _layerBaker.GetInstanceComponentsWithPath(instanceComponents);
+
+    // POC: these are not captured by traversal, so we need to re-add them here
+    if (unpackedRoot.DefinitionProxies != null && unpackedRoot.DefinitionProxies.Count > 0)
+    {
+      var transformed = unpackedRoot.DefinitionProxies.Select(proxy =>
+        (Array.Empty<Collection>(), proxy as IInstanceComponent)
+      );
+      instanceComponentsWithPath.AddRange(transformed);
+    }
+
+    // 3 - Bake materials and colors, as they are used later down the line by layers and objects
+    if (unpackedRoot.RenderMaterialProxies != null)
+    {
+      _materialBaker.ParseAndBakeRenderMaterials(
+        unpackedRoot.RenderMaterialProxies,
+        baseLayerPrefix,
+        onOperationProgressed
+      );
+    }
+
+    if (unpackedRoot.ColorProxies != null)
+    {
+      _colorBaker.ParseColors(unpackedRoot.ColorProxies, onOperationProgressed);
+    }
+
+    PostParseProxies(rootObject, baseLayerPrefix);
```

Comment:
can be removed, clean the property sets with the other deep cleans in step 0

Reply:
Added all the suggestions above in the last commit.